detectionmetrics/utils/metrics.py

```diff
@@ -160,3 +160,51 @@ def get_accuracy(self) -> Tuple[np.ndarray, float]:
         )
         acc = np.sum(correct_per_class) / np.sum(total_per_class)
         return acc_per_class, acc
+
+    def get_precision(self) -> np.ndarray:
+        """Compute precision from confusion matrix. 
+        Precision = true positives/ predicted positives.
+
+        :return: per class precision
+        :rtype: np.ndarray
+        """
+        true_positives = np.diag(self.confusion_matrix)
+        predicted_positives = np.sum(self.confusion_matrix, axis=0)
+        precision_per_class = np.where(
+            predicted_positives > 0, true_positives / predicted_positives, np.nan
+        )
+        global_precision = np.sum(true_positives) / np.sum(predicted_positives)
+        return precision_per_class, global_precision
+
+    def get_recall(self) -> np.ndarray:
+        """Compute recall from Confusion matrix.
+        Recall = true positives / 
+
+        :return: per class recall
+        :rtype: np.ndarray
+        """
+        true_positives = np.diag(self.confusion_matrix)
+        actual_positives = np.sum(self.confusion_matrix, axis=1)
```

Comment:
this gives the total number of samples, in order to get actual positives you would have to subtract actual negatives. In other words, `actual positives = true positives + false negatives`